contracts/base/Dispatcher.sol

```diff
@@ -269,9 +271,40 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, V4SwapRout
                     }
 
                     (success, output) = address(V3_POSITION_MANAGER).call(inputs);
+                } else if (command == Commands.V4_INITIALIZE_POOL) {
+                    bytes4 selector;
+                    assembly {
+                        selector := calldataload(inputs.offset)
+                    }
+                    if (selector != PoolInitializer.initializePool.selector) {
+                        revert InvalidAction(selector);
+                    }
+                    (success, output) = address(V4_POSITION_MANAGER).call(inputs);
                 } else if (command == Commands.V4_POSITION_CALL) {
                     // should only call modifyLiquidities() to mint
-                    // do not permit or approve this contract over a v4 position or someone could use this command to decrease, burn, or transfer your position
+                    bytes4 selector;
+                    assembly {
+                        selector := calldataload(inputs.offset)
+                    }
+                    if (selector != V4_POSITION_MANAGER.modifyLiquidities.selector) {
+                        revert InvalidAction(selector);
+                    }
+
+                    bytes calldata slice = inputs[4:];
+                    (bytes calldata actions,) = slice.toBytes(0).decodeActionsRouterParams();
+
+                    uint256 numActions = actions.length;
+
+                    for (uint256 actionIndex = 0; actionIndex < numActions; actionIndex++) {
+                        uint256 action = uint8(actions[actionIndex]);
+
+                        if (
+                            action == Actions.INCREASE_LIQUIDITY || action == Actions.DECREASE_LIQUIDITY
```

Comment:
i think this should all be in a helper function thats called like `checkV4PositionManagerCall` inside `V3ToV4Migrator`, and then this contract can stay way simpler. And all this logic can just go into there. So here its just

```
} else if (command == Commands.V4_POSITION_CALL) {
    checkV4PositionManagerCall()
    (success, output) = address(V4_POSITION_MANAGER).call{value: address(this).balance}(inputs);
}
```

and i'd argue we shouldve done the same for v3 but that can be separate PR to refactor